docs/topics/proc_exposing-application-metrics-with-prometheus-spring-boot.adoc

```diff
@@ -0,0 +1,165 @@
+= Exposing application metrics using Prometheus in a {SpringBoot} application.
+
+* Configure your application to expose metrics.
+* Collect and view the data using Prometheus.
+
+Prometheus actively connects to a monitored application to collect data; the application does not actively send metrics to a server.
+
+.Prerequisites
+
+* Prometheus configured to collect metrics from the application:
++
+--
+. Download and extract the link:https://prometheus.io/download/[archive^] with the latest Prometheus release:
++
+[source,bash,options="nowrap"]
+----
+$ wget https://github.com/prometheus/prometheus/releases/download/v2.4.3/prometheus-2.4.3.linux-amd64.tar.gz
+$ tar -xvf  prometheus-2.4.3.linux-amd64.tar.gz
+----
+
+. Navigate to the directory with Prometheus:
++
+[source,bash,options="nowrap"]
+----
+$ cd  prometheus-2.4.3.linux-amd64
+----
+
+. Append the following snippet to the `prometheus.yml` file to specify the port where your application exposes the metrics endpoint.
++
+The `scrape_interval` setting is optional and sets Prometheus to collect metrics every 5 seconds.
++
+[source,yaml,options="nowrap"]
+----
+  - job_name: 'spring-actuator'
+    metrics_path: '/actuator/prometheus'
+    scrape_interval: 5s
+    static_configs:
+    - targets: ['localhost:8080']
+----
++
+If you have the `spring-boot-starter-actuator` dependency in your `pom.xml` file, {SpringBoot}-based applications expose metrics at the `/actuator/prometheus` endpoint by default.
+You can change the endpoint at which your application exposes the metrics by setting the `endpoints.prometheus.id` property to a different path, for example: `endpoints.prometheus.id=micrometheus`.
++
+The endpoint is secured by default.
+To ensure that metrics collection works in your application, you must:
+* include authentication in your Prometheus metrics collection endpoint, or
+* make the endpoint unsecured by setting a property, such as `management.security.enabled=false`
+--
+
+* The Prometheus server started on `localhost`:
++
+--
+Start Prometheus and wait until the `Server is ready to receive web requests` message is displayed in the console.
+
+[source,bash,options="nowrap"]
+----
+$ ./prometheus
+----
+--
+
+.Procedure
+
+. Include the `spring-boot-starter-actuator`, `micrometer-registry-prometheus` and `micrometer-spring-legacy` dependencies in your `pom.xml`.
+The `micrometer-spring-legacy` dependency is required to provide Micrometer support for {SpringBoot} 1.5 applications.
++
+.pom.xml
+[source,xml]
+----
+<dependencies>
+  ...
+  <dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-actuator</artifactId>
+  </dependency>
+  <dependency>
+    <groupId>io.micrometer</groupId>
+    <artifactId>micrometer-spring-legacy</artifactId>
+  </dependency>
+  <dependency>
+    <groupId>io.micrometer</groupId>
+    <artifactId>micrometer-registry-prometheus</artifactId>
+  </dependency>
```

Comment:
We need to add the version here. Most likely `1.1.0`

Reply:
Same here, shouldn't it come from the BOM?

Reply:
It would if we had those deps in the BOM :P